### PR TITLE
92 2j add poi

### PR DESCRIPTION
**Pull request by claraxuxu**

### CODE REVIEW

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 1]
```diff
@@ -1,12 +1,38 @@
 package com.bouboulecorp.thurii.HomePage.HomePageFragments
+import android.Manifest
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:2]: 🚫 **[ktlint]** <import-ordering> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Imports must be ordered in lexicographic order without any empty lines in-between with "java", "javax", "kotlin" and aliases in the end

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+import android.content.pm.PackageManager
+import android.graphics.BitmapFactory
+import com.google.android.gms.location.LocationRequest
 import android.os.Bundle
+import android.os.PersistableBundle
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:7]: 🚫 **[ktlint]** <no-unused-imports> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 3]
```diff
@@ -19,9 +45,114 @@ class MapFragment : Fragment() {
...
+                            literal(TRASH_ICON_ID)
+                        }
+                    )
+                    iconSize(0.15)
+                    iconAllowOverlap(false)
+                    iconAnchor(IconAnchor.BOTTOM)
+                }
+            }
+        )
+        val filterButtonList = arrayListOf<map_recyclerview_item>() //list of buttons
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:101]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 4]
```diff
@@ -19,9 +45,114 @@ class MapFragment : Fragment() {
...
+        filterButtonList.add(map_recyclerview_item("climbing_spot", R.drawable.ic_climbing))
+        filterButtonList.add(map_recyclerview_item("fishing_spot", R.drawable.ic_fishing))
+        filterButtonList.add(map_recyclerview_item("danger", R.drawable.ic_forbiden))
+        filterButtonList.add(map_recyclerview_item("water_spot", R.drawable.ic_water))
+        filterButtonList.add(map_recyclerview_item("landscape", R.drawable.ic_camera))
+        filterButtonList.add(map_recyclerview_item("trash", R.drawable.ic_trash))
+
+        val horizontalRecyclerView = view.findViewById<RecyclerView>(R.id.horizontal_caroussel)
+        horizontalRecyclerView.adapter = map_recyclerview_item_adapter(
+            filterButtonList, R.layout.item_on_board, requireContext()
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:114]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:114]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 5]
```diff
@@ -19,9 +45,114 @@ class MapFragment : Fragment() {
...
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        mapView?.onDestroy()
+    }
+
+    private val locationPermissionCode = 2
+    private fun startLocationUpdates() {
+        if ((ContextCompat.checkSelfPermission(
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:141]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline after "("

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 6]
```diff
@@ -19,9 +45,114 @@ class MapFragment : Fragment() {
...
+                requireContext(),
+                Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED)
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:143]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline before ")"

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:143]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline before ")"

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:143]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline before ")"

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 7]
```diff
@@ -19,9 +45,114 @@ class MapFragment : Fragment() {
...
+        ) {
+            ActivityCompat.requestPermissions(
+                requireActivity(),
+                arrayOf(Manifest.permission.ACCESS_FINE_LOCATION),
+                locationPermissionCode
+            )
+        }
+        else {
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:151]: 🚫 **[ktlint]** <keyword-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected newline before "else"

File: app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt [hunk 8]
```diff
@@ -0,0 +1,33 @@
+package com.bouboulecorp.thurii.data
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt:1]: 🚫 **[ktlint]** <final-newline> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>File must end with a newline (\n)

File: app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageButton
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.recyclerview.widget.RecyclerView
+import com.bouboulecorp.thurii.R
+
+class map_recyclerview_item_adapter(
+    private val filterButtonList: List<map_recyclerview_item>, private val layoutId: Int, private val cont : Context
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt:13]: 🚫 **[ktlint]** <parameter-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Parameter should be on a separate line (unless all parameters can fit a single line)

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt:13]: 🚫 **[ktlint]** <parameter-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Parameter should be on a separate line (unless all parameters can fit a single line)

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt:13]: 🚫 **[ktlint]** <colon-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected spacing before ":"

File: app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+) : RecyclerView.Adapter<map_recyclerview_item_adapter.ViewHolder>()
+{
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt:15]: 🚫 **[ktlint]** <curly-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected newline before "{"

File: app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        val button : ImageButton = view.findViewById(R.id.imageButtonItem)
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt:17]: 🚫 **[ktlint]** <colon-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected spacing before ":"

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 12]
```diff
@@ -19,9 +44,118 @@ class MapFragment : Fragment() {
...
+
+    override fun onDestroy() {
+        super.onDestroy()
+        mapView?.onDestroy()
+    }
+
+    private val locationPermissionCode = 2
+    private fun startLocationUpdates() {
+        if (
+            (ContextCompat.checkSelfPermission(
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:143]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline after "("

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 13]
```diff
@@ -19,9 +44,118 @@ class MapFragment : Fragment() {
...
+                requireContext(),
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED
+                    )
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:147]: 🚫 **[ktlint]** <indent> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected indentation (20) (should be 16)

File: app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+import android.widget.ImageButton
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.recyclerview.widget.RecyclerView
+import com.bouboulecorp.thurii.R
+
+class map_recyclerview_item_adapter(
+    private val filterButtonList: List<map_recyclerview_item>,
+    private val layoutId: Int,
+    private val cont: Context
+): RecyclerView.Adapter<map_recyclerview_item_adapter.ViewHolder>() {
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt:16]: 🚫 **[ktlint]** <colon-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing spacing before ":"

File: app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+    class ViewHolder(view: View): RecyclerView.ViewHolder(view) {
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/data/map_recyclerview_item_adapter.kt:17]: 🚫 **[ktlint]** <colon-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing spacing before ":"

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 16]
```diff
@@ -19,9 +44,119 @@ class MapFragment : Fragment() {
...
+        super.onDestroy()
+        mapView?.onDestroy()
+    }
+
+    private val locationPermissionCode = 2
+    private fun startLocationUpdates() {
+        if (
+            (
+                ContextCompat.checkSelfPermission(
+                requireContext(),
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:145]: 🚫 **[ktlint]** <indent> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected indentation (16) (should be 20)

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 17]
```diff
@@ -19,9 +44,119 @@ class MapFragment : Fragment() {
...
+                Manifest.permission.ACCESS_FINE_LOCATION
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:146]: 🚫 **[ktlint]** <indent> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected indentation (16) (should be 20)

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 18]
```diff
@@ -19,9 +44,119 @@ class MapFragment : Fragment() {
...
+            ) != PackageManager.PERMISSION_GRANTED
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:147]: 🚫 **[ktlint]** <indent> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected indentation (12) (should be 16)

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 19]
```diff
@@ -19,9 +44,119 @@ class MapFragment : Fragment() {
...
+
+    private val locationPermissionCode = 2
+    private fun startLocationUpdates() {
+        if (
+            (
+                ContextCompat.checkSelfPermission(
+                    requireContext(),
+                    Manifest.permission.ACCESS_FINE_LOCATION
+                ) != PackageManager.PERMISSION_GRANTED
+            )
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:148]: 🚫 **[ktlint]** <indent> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected indentation (12) (should be 16)

**Review by larroyo3 — CHANGES_REQUESTED** [review]
il me semble que la bar pour filtrer les POI n'est pas à faire sur cette PR

tu peux la garder de coté pour ta prochaine issue "Filter POI" mais je pense c'est mieux de pas mélanger les issues et de l'enlever de celle ci

**Comment by claraxuxu** [comment]
> il me semble que la bar pour filtrer les POI n'est pas à faire sur cette PR
> 
> tu peux la garder de coté pour ta prochaine issue "Filter POI" mais je pense c'est mieux de pas mélanger les issues et de l'enlever de celle ci

t'as bien raison, je viens de push une version propre

**Review by larroyo3 — CHANGES_REQUESTED** [review]
Est-ce que c'est possible d'avoir les icon en svg plutot qu'en png ? 
Sinon c'est tout bon pour moi :+1:

**Comment by claraxuxu** [comment]
- [ ] les îcons de POI en formate texte(emoji) arrivent pas à être exporter correctement à format SVG

**Pull request merged**